ios/uPortMobile/Classes/UPTEthSigner.m

```diff
@@ -200,8 +84,13 @@ + (void)signTransaction:(NSString *)ethAddress data:(NSString *)payload userProm
     if (key) {
         NSData *payloadData = [[NSData alloc] initWithBase64EncodedString:payload options:0];
         NSData *hash = [UPTEthSigner keccak256:payloadData];
-        NSDictionary *signature = [self ethereumSignature: key forHash:hash];
-        result(signature, nil);
+        NSDictionary *signature = ethereumSignature(key, hash, NULL);
+        if (signature) {
+            result(signature, nil);
+        } else {
+            NSError *signingError = [[NSError alloc] initWithDomain:@"UPTError" code:UPTSignerErrorCodeLevelSigningFailed.integerValue userInfo:@{@"message": @"signing failed due to invalid values"}];
```

Comment:
I think it's good to keep lines shorter than 120 characters. Please break up this line, if needed by introducing local variables:

 ```
NSError *signingError;
signingError = [[NSError alloc] initWithDomain:@"UPTError" 
                                                 code:UPTSignerErrorCodeLevelSigningFailed.integerValue
                                                  userInfo:@{@"message": @"signing failed due to invalid values"}];
```